llmfoundry/utils/exceptions.py

```diff
@@ -35,9 +35,17 @@
 ALLOWED_PROMPT_KEYS = {'prompt'}
 ALLOWED_MESSAGES_KEYS = {'messages'}
 
+ErrorContext = Union[Literal['TrainContext'], Literal['EvalContext']]
```

Comment:
Thoughts on calling this generally ModelTrainingError with the property "location"  = "train dataloader" || "eval dataloader"

and then we could extend this to catching more errors with metadata in the future!

Reply:
I think `ModelTrainingError` might not be general enough because we'd want this in our eval script too, and maybe also in our data prep scripts. I was thinking initially we could drop the "Training" and call it a `ModelError`, but that's even more misleading because it implies the issue is with the model and not the train/eval dataloaders. I ended up sticking with ContextualError, because it seemed like the most general way to describe an error that could happen anywhere in the codebase and contained a context, but changed the `context` attribute to `location` to make the situation clearer.